command/operator_debug.go

```diff
@@ -460,51 +488,298 @@ func (c *OperatorDebugCommand) Run(args []string) int {
 	return 0
 }
 
-// collect collects data from our endpoints and writes the archive bundle
-func (c *OperatorDebugCommand) collect(client *api.Client) error {
-	// Version contains cluster meta information
-	dir := "version"
+// targetMap generic map for targets
+type targetMap map[string]bool
 
-	self, err := client.Agent().Self()
-	c.writeJSON(dir, "agent-self.json", self, err)
+// String returns a CSV of valid enabled targets
+func (t targetMap) String() string {
+	var targets []string
+	var targetString string
+
+	for target, enabled := range t {
+		if enabled && target != "all" {
+			targets = append(targets, target)
+		}
+	}
+	sort.Strings(targets)
+	targetString = strings.Join(targets, ",")
+
+	return targetString
+}
+
+// IsValid checks whether the key exists in the map
+func (m targetMap) IsValid(value string) bool {
+	if value == "all" {
+		return true
+	}
+	_, ok := m[value]
+	return ok
+}
+
+// parseTargets parses a comma delimited target string into a targetMap
+func (c *OperatorDebugCommand) parseTargets(targets string) targetMap {
+	ret := make(targetMap)
+
+	// Build starting target map
+	switch {
+	case targets == "":
+		return ret
```

Comment:
Should we return `targetDefaultsNone()` here? If we return an empty uninitialized `targetMap`, I fear that follow up calls with `IsValid` will generate noisy invalid keys messages because the map is empty.